### DISCUSSION TITLE
Glances running in systemclt ad server and exporter

### DISCUSSION THREAD

**Discussion by felipebraz** [nicolargo/glances#2677] [Q&A] [answered]
I'm already running glances as a service in systemctl:

[Unit]
Description=Glances
After=network.target

[Service]
ExecStart=/usr/local/bin/glances -w -u glances
Restart=on-abort
RemainAfterExit=yes

[Install]
WantedBy=multi-user.target

I want to run it as a exporter to InfluxDB with the service (webserver and exporter at the same time): glances -t 5 --export influxdb2

Is there a way to do it?

Sorry if the question is a little newbie, I'm a little newbie. LoL

**Comment by nicolargo** [comment]
Hi @felipebraz 

there is no newbie question :)

If you want to use the WebUI and the export to InfluxDB2 at the same time, you need to create two services.

One for the WebUI:

```
[Unit]
Description=Glances
After=network.target

[Service]
ExecStart=/usr/local/bin/glances -w -u glances
Restart=on-abort
RemainAfterExit=yes

[Install]
WantedBy=multi-user.target
```

And another one for the exporter:

```
[Unit]
Description=Glances Exporter
After=network.target

[Service]
ExecStart=/usr/local/bin/glances -t 5 --export influxdb2
Restart=on-abort
RemainAfterExit=yes

[Install]
WantedBy=multi-user.target
```

For your information, there is an open feature request in the Glances Roadmap waiting for a contributor: https://github.com/nicolargo/glances/issues/1527

**Reply by felipebraz** [reply]
I've tried your suggestion but I'm receiving an error:

root@prox:/usr/local/bin# systemctl status glancesExport.service
● glancesExport.service - Glances Exporter
     Loaded: loaded (/etc/systemd/system/glancesExport.service; enabled; preset: enabled)
     Active: active (running) since Sun 2024-02-11 23:37:24 -03; 3s ago
   Main PID: 170142 (glances)
      Tasks: 2 (limit: 38233)
     Memory: 49.1M
        CPU: 1.242s
     CGroup: /system.slice/glancesExport.service
             └─170142 /usr/bin/python3 /usr/local/bin/glances -t 5 --export influxdb2

Feb 11 23:37:25 prox glances[170142]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Feb 11 23:37:25 prox glances[170142]:   File "/usr/local/lib/python3.11/dist-packages/glances/standalone.py", line 106, in __init__
Feb 11 23:37:25 prox glances[170142]:     self.screen = GlancesCursesStandalone(config=config, args=args)
Feb 11 23:37:25 prox glances[170142]:                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Feb 11 23:37:25 prox glances[170142]:   File "/usr/local/lib/python3.11/dist-packages/glances/outputs/glances_curses.py", line 143, in __init__
Feb 11 23:37:25 prox glances[170142]:     self.screen = curses.initscr()
Feb 11 23:37:25 prox glances[170142]:                   ^^^^^^^^^^^^^^^^
Feb 11 23:37:25 prox glances[170142]:   File "/usr/lib/python3.11/curses/__init__.py", line 29, in initscr
Feb 11 23:37:25 prox glances[170142]:     setupterm(term=_os.environ.get("TERM", "unknown"),
Feb 11 23:37:25 prox glances[170142]: _curses.error: setupterm: could not find terminal
root@prox:/usr/local/bin# systemctl status glancesExport.service
× glancesExport.service - Glances Exporter
     Loaded: loaded (/etc/systemd/system/glancesExport.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Sun 2024-02-11 23:37:29 -03; 9s ago
   Duration: 5.528s
    Process: 170142 ExecStart=/usr/local/bin/glances -t 5 --export influxdb2 (code=exited, status=1/FAILURE)
   Main PID: 170142 (code=exited, status=1/FAILURE)
        CPU: 1.306s

If I run the command (/usr/local/bin/glances -t 5 --export influxdb2) directly on shell it loads the glances screen and exports just fine.

**Reply by nicolargo** [reply, marked as answer]
Add the --quiet  option to diable the UI.

**Comment by felipebraz** [comment]
Thank you very much for your patience and help!

**Comment by felipebraz** [comment]
Thank you very much again!
